File: 백준/Silver/10826. 피보나치 수 4/피보나치 수 4.java
import java.awt.List;
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.math.BigInteger;

public class Main{
	public static void main (String [] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		BigInteger [] arr = new BigInteger [10001];
		arr[0] = BigInteger.ZERO;
		arr[1] = BigInteger.ONE;
		for (int i = 2; i < arr.length; i++) {
			arr[i] = arr[i-2].add(arr[i-1]);
		}
		int N = Integer.parseInt(br.readLine());
		System.out.println(arr[N]);
	}
	}

File: 백준/Silver/10826. 피보나치 수 4/피보나치 수 4.py
n = int(input())
arr= [0] * (n+1)
if n:
    arr[1] = 1

    for i in range(2, n+1):
        arr[i] = arr[i-1] + arr[i-2]

    print(arr[n])
else:
    print(0)